doc/_extensions/odoo_ext/translator.py

```diff
@@ -515,6 +515,20 @@ def visit_reference(self, node):
         self.body.append(self.starttag(node, 'a', **attrs))
     def depart_reference(self, node):
         self.body.append(u'</a>')
+    def visit_download_reference(self, node):
+        # type: (nodes.Node) -> None
+        if node.hasattr('filename'):
+            self.body.append(
+                '<a class="reference download internal" href="%s" download="">' %
+                posixpath.join(self.builder.dlpath, node['filename']))
+            self.body.append(node.astext())
+            self.body.append('</a>')
+            raise nodes.SkipNode
```

Comment:
Why not add the end node to the `context` and let normal processing take care of the children?

Reply:
I didn't code it myself. It is a copy/paste from [https://github.com/odoo/documentation-user/blob/14.0/_extensions/odoo/translator.py](https://github.com/odoo/documentation-user/blob/14.0/_extensions/odoo/translator.py) @AntoineVDV  sent me.